analyses/cell-type-wilms-tumor-06/notebook/04_annotation_Across_Samples_exploration.Rmd

```diff
@@ -0,0 +1,310 @@
+  summarise(Nb = n())  %>%
+  mutate(C = sum(Nb)) %>%
+  mutate(percent = Nb/C*100)
+
+  
+p1 <- ggplot(df, aes(x= !!sym(first_group), y = !!sym(last_group), fill = Nb)) +
+   geom_tile() +
+   scale_fill_viridis_c()  +
+   theme_bw() +
+   theme(text = element_text(size = 20)) +
+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
+   guides(fill=guide_legend(title=" Number of \n cells"))
+
+p2 <- ggplot(df, aes(x= !!sym(first_group), y = !!sym(last_group), fill = percent)) +
+   geom_tile() +
+   scale_fill_viridis_c()  +
+   theme_bw() +
+   theme(text = element_text(size = 20)) +
+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
+   guides(fill=guide_legend(title=" Percentage of \n cells"))
+
+p <- p1 + p2 + plot_layout(ncol = 2)
+return(p)
+}
+```
+
+#### do_Feature_mean
+
+`do_Feature_mean` shows heatmap of mean expression of a feature grouped by a metadata. 
+
+- `group.by` is the name of the metadata to group the cells
+- `feature` is the name of the gene to average the expression
+
+```{r fig.width=10, fig.height=4, out.width='100%'}
+do_Feature_mean <- function(group.by, feature ){
+  
+  df <- cell_type_df %>%
+  group_by(sample_id, !!sym(group.by))%>% 
+  summarise(m = mean(!!sym(feature)))  
+
+  
+p <- ggplot(df, aes(x= sample_id, y = !!sym(group.by), fill = m)) +
+   geom_tile() +
+   scale_fill_viridis_c()  +
+   theme_bw() +
+   theme(text = element_text(size = 20)) +
+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
+   guides(fill=guide_legend(title=paste0(feature)))
+
+return(p)
+}
+```
+
+## Analysis
+
+### Total number of cells per samples
+
+```{r fig.width=5, fig.height=10, out.width='100%', results='asis'}
+tmp <- table(cell_type_df$sample_id)
+knitr::kable(tmp)
+```
+### Predicted organ
+
+The predicted organ comes from label transfer from the human fetal atlas. 
+This is the reference developed by [Cao et al.](https://www.science.org/doi/10.1126/science.aba7721) provided by `Azimuth` for label transfer.
+The reference contain cells from 15 organs including kidney from fetal samples.
+The label transfer have been performed in the notebook `02a_fetal_full_reference_Cao_{sample_id}.Rmd`
+
+The fetal full reference (Cao et al.) provides three levels of annotations:
+
+- `fetal_full_predicted.organ` is one of the 15 organs used for building the reference. 
+We were nicely surprised that, for many of the samples, most of the cells from the Wilms tumor cohort mapped to fetal kidney cells. 
+
+- `fetal_full_predicted.annotation.l1` holds predicted cell type annotation
+
+- `fetal_full_predicted.annotation.l2` holds the combination of the predicted organ and predicted cell type annotation
+
+    
+
+```{r fig.width=25, fig.height=6,  results='asis'}
+tmp <- table(cell_type_df$sample_id, cell_type_df$organ)
+
+knitr::kable(tmp)
+```
+
+```{r fig.width=25, fig.height=6}
+do_Table_Heatmap(first_group = "sample_id", last_group = "organ")
+```
```

Comment:
We can delete these chunks and replace with this code to make a table of counts for kidney vs not kidney. Then, style the data table however you usually like to do!

This table will have 3 columns in addition to the sample id: a total cell count, the percentage of cells labeled kidney, the percentage of cells labeled anything else that isn't kidney. This will make it easy for us to see which samples do really well, and which don't, for this reference.

```
kidney_count_df <- cell_type_df |>
  # make a new variable that tells us if it's kidney or not
  dplyr::mutate(organ_type = ifelse(organ == "Kidney", "kidney", "not_kidney"))  |>
  # count how many kidney or not
  dplyr::count(sample_id, organ_type) |>
  # make the data frame wide and use a value of 0 if a count is missing
  tidyr::pivot_wider(names_from = organ_type, values_from = n, values_fill=n) |>
  # add a column for total, and convert other counts to percentages
  dplyr::mutate(
    total = kidney + not_kidney, 
    kidney = round(kidney/total, 4)*100, 
    not_kidney = round(not_kidney/total, 4)*100) |>
  # arrange in descending order of percentage of kidney cells
  dplyr::arrange(desc(kidney)) 

DT::datatable(kidney_count_df)
```

Reply:
This is great thank you. I didn't know the function `pivot_wider`, that is great!
It is now running with `values_fill=0` instead of `values_fill=n`, if I understood correctly the help of the function and your code :)